common.js

```diff
@@ -8,13 +9,33 @@ const VERSIONS_JSON = 'https://ziglang.org/download/index.json';
 const MACH_VERSIONS_JSON = 'https://pkg.machengine.org/zig/index.json';
 const CACHE_PREFIX = "setup-zig-global-cache-";
 
+const MINIMUM_ZIG_VERSION_REGEX = /\.minimum_zig_version\s*=\s*"(.*?)"/;
+
 let _cached_version = null;
 async function getVersion() {
   if (_cached_version != null) {
     return _cached_version;
   }
 
-  const raw = core.getInput('version');
+  let raw = core.getInput('version');
+  if (raw === '') {
+    try {
+      const zon = await fs.promises.readFile('build.zig.zon', 'utf8');
+      const match = MINIMUM_ZIG_VERSION_REGEX.exec(zon);
```

Comment:
Potential future enhancement: I believe the Mach project is looking at including a `mach_zig_version` field in their `build.zig.zon`? So we could also look for that field and interpret it as a Mach version.